st2api/st2api/controllers/v1/keyvalue.py

```diff
@@ -61,19 +63,22 @@ def __init__(self):
         self.get_one_db_method = self._get_by_name
 
     @jsexpose(arg_types=[str, str, str, bool])
-    def get_one(self, name, scope=SYSTEM_SCOPE, user=None, decrypt=False):
+    def get_one(self, name, scope=None, user=None, decrypt=False):
         """
             List key by name.
 
             Handle:
                 GET /keys/key1
         """
-        self._validate_scope(scope=scope)
+        if not scope:
+            scope = SYSTEM_SCOPE
```

Comment:
Yeah, I would prefer to do both if we have to - default in the method signature and also leave this part here.

And in short I don't think it's better, we should always assign default values to the arguments in the function signature (that's how it was designed and supposed to be used), but I assume it has something to do with the way pecan handles argument so the actual default argument value is overridden with `None` when calling the method.

In any case, as mentioned above, I would still prefer to do both in such scenario - specify default argument value in the method signature and do additional default assignment in the method body, if needed.